src/main/java/core/mate/academy/service/BulldozerProducer.java

```diff
@@ -0,0 +1,20 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Machine;
+import java.util.ArrayList;
+import java.util.List;
+
+public class BulldozerProducer implements MachineProducer<Machine> {
```

Comment:
The `MachineProducer` interface should be parameterized with `Bulldozer` to prevent the creation of a producer that does not use `Machine` or its subtype. Change `MachineProducer<Machine>` to `MachineProducer<Bulldozer>`.